salt/utils/json.py

```diff
@@ -16,6 +16,19 @@
 JSONEncoder = json.JSONEncoder
 
 
+class DatetimeEncoder(json.JSONEncoder):
+    """Datetime encoder to handle json serialization"""
+
+    def default(self, obj):
+        try:
+            return super().default(obj)
+        except TypeError as e:
+            if isinstance(obj, (datetime.date, datetime.datetime)):
+                return obj.isoformat()
+
+            raise TypeError(e)
```

Comment:
```suggestion
    def default(self, obj):
        if isinstance(obj, (datetime.date, datetime.datetime)):
            return obj.isoformat()
        return super().default(obj)
```
I'd prefer something like this.